auth/jwtverifier/insecure.go

```diff
@@ -0,0 +1,83 @@
+package jwtverifier
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+	"sync"
+
+	"gopkg.in/square/go-jose.v2/jwt"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// InsecureVerifier parses JWTs from the Authorization header WITHOUT signature verification.
+// This mode is ONLY for development and testing. It requires the ALLOW_INSECURE_JWT=true
+// environment variable to be set as a safety check.
+//
+// WARNING: Never use this in production - it accepts any JWT regardless of signature.
+type InsecureVerifier struct {
+	warnedOnce sync.Once
+}
+
+// NewInsecureVerifier creates a new insecure JWT verifier.
+// Returns an error if the ALLOW_INSECURE_JWT environment variable is not set to "true".
+func NewInsecureVerifier() (*InsecureVerifier, error) {
+	// Require explicit environment variable as safety check
+	if os.Getenv("ALLOW_INSECURE_JWT") != "true" {
+		return nil, fmt.Errorf("insecure JWT mode requires ALLOW_INSECURE_JWT=true environment variable")
+	}
+
+	log.Warn("======================================================================")
+	log.Warn("INSECURE JWT MODE ENABLED - JWTs will NOT be validated!")
+	log.Warn("This mode should ONLY be used in development/testing environments")
+	log.Warn("DO NOT USE IN PRODUCTION")
+	log.Warn("======================================================================")
+
+	return &InsecureVerifier{}, nil
+}
+
+// ExtractClaims extracts JWT claims from the Authorization header without signature verification.
+func (v *InsecureVerifier) ExtractClaims(req *http.Request) (map[string]interface{}, error) {
+	// Log warning on first use (per-verifier instance)
+	v.warnedOnce.Do(func() {
+		log.Warn("INSECURE MODE: Parsing JWT without signature verification")
+	})
+
+	// Extract Authorization header
+	authHeader := req.Header.Get("Authorization")
+	if authHeader == "" {
+		return nil, fmt.Errorf("Authorization header not found")
+	}
+
+	if !strings.HasPrefix(authHeader, "Bearer ") {
+		return nil, fmt.Errorf("Authorization header must be in format: Bearer <token>")
+	}
+
+	tokenString := strings.TrimPrefix(authHeader, "Bearer ")
+
+	// Parse JWT without verification
```

Comment:
🤔 The comment says "Parse JWT without verification" but the code is the same as for `direct.go` and invokes the same `jwt.ParseSigned` function. So, I think it's a bit misleading. I think the part where you lack verification is where you extract claims, which is clearly marked below as insecure (good!).